### DISCUSSION TITLE
Extract IFC class instances using IfcConvert CLI?

### DISCUSSION THREAD

**Discussion by FransKnibbe** [IfcOpenShell/IfcOpenShell#4541] [Q&A] [answered]
Hello! I am looking for a way to extract individual IFC objects from an IFC file into GLB files. For example: convert one IfcSpace instance with a certain name or ID to a GLB file. I have looked at the include and exclude arguments, but so far my attempts have been unsuccessful.

**Comment by Moult** [comment]
Gday! Have you read https://docs.ifcopenshell.org/ifcconvert/usage.html ?

Dion Moult

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by FransKnibbe** [comment]
Yes, I have read the man page. Based on that, I tried something like

IfcConvert --include=Name "spaceX" in.ifc spaceX.glb

That results in a syntax error:

[Error] Invalid usage of '--include': the argument for option '--include' is invalid

So I think I actually have two questions: 
1) How can I get the syntax for the --include' argument right?
2) Does --include exclude everything else?

**Reply by Moult** [reply, marked as answer]
The syntax is described perhaps it's not too clear without examples. Here's an example:

```
./IfcConvert  -v --include attribute GlobalId 1yETHMphv6LwABqR4Pbs5g attribute GlobalId 1yETHMphv6LwABqR0Pbs5g model.ifc out.glb
```

Yes, if you use include, then everything else is excluded.

**Comment by FransKnibbe** [comment]
Thanks a lot, Dion! The example certainly helped. At first I tried something like your example, but then I got the error
`[Error] Input file not specified`. Then I noticed this bit in the usage text: "-include and --exclude  cannot be placed right before input                                         file argument". Moving the -v argument helped to resolve that. So somehting like this works for me:
`./IfcConvert --include entities IfcSpace attribute GlobalId 0b1r3gEG57G93Rt29P2yO$ -v model.ifc out.glb`.
